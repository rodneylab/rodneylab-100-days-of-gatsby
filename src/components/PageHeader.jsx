import React from 'react';
import { graphql, useStaticQuery } from 'gatsby';
import { Helmet } from 'react-helmet';
import PropTypes from 'prop-types';

import { VERTICAL_LINE_ENTITY } from '../constants/entities';

export const PurePageHeader = ({
  data: {
    site: {
      siteMetadata: { title, siteLanguage },
    },
  },
  pageTitle,
}) => (
  <>
    <Helmet
      title={`${title} ${VERTICAL_LINE_ENTITY} ${pageTitle}`}
      htmlAttributes={{ lang: siteLanguage }}
    />
    <Helmet>
      <meta name="description" content={`${title} ${pageTitle} page`} />
      <meta
        name="robots"
        content="index, follow, max-snippet:-1, max-image-preview:large, max-video-preview:-1"
      />
    </Helmet>
  </>
);

PurePageHeader.propTypes = {
  data: PropTypes.shape({
    site: PropTypes.shape({
      siteMetadata: PropTypes.shape({
        title: PropTypes.string,
        siteLanguage: PropTypes.string,
      }),
    }),
  }).isRequired,
  pageTitle: PropTypes.string.isRequired,
};

export const query = graphql`
  fragment PageHeaderFragment on Site {
    siteMetadata {
      title
      siteLanguage
    }
  }
`;

const PageHeader = ({ pageTitle }) => {
  const data = useStaticQuery(
    graphql`
      query PageHeaderQuery {
        site {
          ...PageHeaderFragment
        }
      }
    `,
  );
  return <PurePageHeader data={data} pageTitle={pageTitle} />;
};

PageHeader.propTypes = {
  pageTitle: PropTypes.string.isRequired,
};

export { PageHeader as default };
