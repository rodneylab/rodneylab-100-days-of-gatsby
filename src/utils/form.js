import { createClient } from '@formium/client';

export const formium = createClient(process.env.GATSBY_FORMIUM_PROJECTID);

export const validate = (values) => {
  const errors = {};
  if (!values.name) {
    errors.name = 'Please enter your name.';
  }
  if (!values.email) {
    errors.email = 'Please enter your email address.';
  } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
    errors.email = 'Please check your email address.';
  }
  if (!values.message) {
    errors.message = 'Please enter a message.';
  } else if (values.message.length > 1024) {
    errors.message = 'Please limit your message to 1000 characters.';
  }
  return errors;
};
