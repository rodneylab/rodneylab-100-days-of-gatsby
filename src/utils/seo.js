import { getSrc } from 'gatsby-plugin-image';

export const getFeaturedImage = ({ featuredImage }) => ({
  url: getSrc(featuredImage.localFile),
  width: 992,
  height: 730,
});

export const getOgImage = ({ ogImage }) => ({
  url: ogImage ? getSrc(ogImage.localFile) : null,
  width: 1200,
  height: 627,
});

export const getOgSquareImage = ({ ogSquareImage }) => ({
  url: ogSquareImage ? getSrc(ogSquareImage.localFile) : null,
  width: 400,
  height: 400,
});

export const getTwitterImage = ({ twitterImage }) => ({
  url: twitterImage ? getSrc(twitterImage.localFile) : null,
  width: 800,
  height: 418,
});
