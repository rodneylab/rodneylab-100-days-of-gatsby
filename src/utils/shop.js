const formatPrice = (currency, value) => Intl.NumberFormat(undefined, {
  currency,
  minimumFractionDigits: 2,
  style: 'currency',
}).format(value);

export { formatPrice as default };

export const localStorageKey = 'a856f4';
