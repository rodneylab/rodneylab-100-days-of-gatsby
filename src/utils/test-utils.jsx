import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';
import { createMemorySource, createHistory, LocationProvider } from '@reach/router';

const source = createMemorySource('active');
const history = createHistory(source);

const locationProvider = ({ children }) => (
  <LocationProvider history={history}>{children}</LocationProvider>
);

export const customRender = (ui, options) => render(ui, { wrapper: locationProvider, ...options });

export const getMetaByName = (metaName) => {
  const metas = document.getElementsByTagName('meta');
  for (let i = 0; i < metas.length; i += 1) {
    if (metas[i].getAttribute('name') === metaName) {
      return metas[i].getAttribute('content');
    }
  }
  return '';
};

export const getMetaByProperty = (metaProperty) => {
  const metas = document.getElementsByTagName('meta');
  for (let i = 0; i < metas.length; i += 1) {
    if (metas[i].getAttribute('property') === metaProperty) {
      return metas[i].getAttribute('content');
    }
  }
  return '';
};
