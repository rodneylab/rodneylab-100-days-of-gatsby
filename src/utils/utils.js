// eslint-disable-next-line import/prefer-default-export
export const isBrowser = typeof window !== 'undefined';
