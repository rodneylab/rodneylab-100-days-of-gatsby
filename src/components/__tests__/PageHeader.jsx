import React from 'react';
import { waitFor } from '@testing-library/react';

import { data, pageTitle } from './__fixtures__/pageHeader';
import { PurePageHeader as PageHeader } from '../PageHeader';
import { customRender } from '../../utils/test-utils';

const elementToRender = (
  <>
    <PageHeader data={data} pageTitle={pageTitle} />
    <h1>Test heading</h1>
    <p>Test body</p>
  </>
);

describe('Header', () => {
  it('renders correctly', async () => {
    const { asFragment } = customRender(elementToRender);
    expect(asFragment(elementToRender)).toMatchSnapshot();
  });

  it('creates html lang tag', async () => {
    customRender(elementToRender);
    await waitFor(() => {
      const htmlTag = document.getElementsByTagName('html')[0];
      expect(htmlTag.getAttribute('lang')).toEqual(data.site.siteMetadata.siteLanguage);
    });
  });

  it('creates head title element', async () => {
    customRender(elementToRender);
    await expect(document.title).toEqual(`${data.site.siteMetadata.title} | ${pageTitle}`);
  });

  it('creates a robot meta tag', async () => {
    customRender(elementToRender);
    await waitFor(() => {
      const tags = document.getElementsByTagName('meta');
      const filteredTags = [].slice.call(tags).filter((tag) => tag.getAttribute('name') === 'robots');
      expect(filteredTags.length).toBe(1);
      expect(filteredTags[0].getAttribute('content')).toBe('index, follow, max-snippet:-1, max-image-preview:large, max-video-preview:-1');
    });
  });
});
