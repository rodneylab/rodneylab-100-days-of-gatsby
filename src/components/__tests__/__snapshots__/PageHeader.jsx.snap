// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<DocumentFragment>
  <h1>
    Test heading
  </h1>
  <p>
    Test body
  </p>
</DocumentFragment>
`;
